cli exits crashes with nil pointer panic when error occurs during push to remote registry

**What happened**:
the cli crashed with a nil pointer panic during erronous push of a component archive to a remote registry after signing.

in my case, the registry returned a 401 Unauthorized error, and it seems that the call to `errors.As()`  panicked (see here: https://github.com/open-component-model/ocm/blob/main/pkg/docker/pusher.go#L194 ).

stacktrace:

```
Starting: /Users/I500806/go/bin/dlv dap --check-go-version=false --listen=127.0.0.1:52753 --log-dest=3 from /Users/I500806/arbeit/ocm/cmds/ocm
DAP server listening at: 127.0.0.1:52753
Type 'dlv help' for list of commands.
applying to version "github.com/test:v0.1.0"...
  resource 0:  "name"="img": digest SHA-256:e7c8624da390331f5d1e4b1bacf9eb78c5e41dba5c3ef7fcbe39cb19906e1bf9[genericBlobDigest/v1]
successfully signed github.com/test:v0.1.0 (digest SHA-256:5d7a7f18631820562ab3a7598aa2566e9c4214a40e2f937f8a34ed9d4b553b3c)
panic: errors: target must be a non-nil pointer

goroutine 1 [running]:
errors.As({0x39480e0, 0xc000a55040}, {0x3612ba0, 0xc000a55090})
	/usr/local/opt/go/libexec/src/errors/wrap.go:85 +0x425
github.com/pkg/errors.As({0x39480e0, 0xc000a55040}, {0x3612ba0, 0xc000a55090})
	/Users/I500806/go/pkg/mod/github.com/pkg/errors@v0.9.1/go113.go:31 +0x46
github.com/open-component-model/ocm/pkg/docker.dockerPusher.push({0xc0001b6640, {0x0, 0x0}, {0xe720fc0, 0xc0004df320}}, {0x3963ee0, 0xc0007392f0}, {{0x37ba313, 0x46}, {0xc00093ceb0, ...}, ...}, ...)
	/Users/I500806/arbeit/ocm/pkg/docker/pusher.go:206 +0x2d31
github.com/open-component-model/ocm/pkg/docker.dockerPusher.Push({0xc0001b6640, {0x0, 0x0}, {0xe720fc0, 0xc0004df320}}, {0x3963ee0, 0xc000141bc0}, {{0x37ba313, 0x46}, {0xc00093ceb0, ...}, ...}, ...)
	/Users/I500806/arbeit/ocm/pkg/docker/pusher.go:49 +0x173
github.com/open-component-model/ocm/pkg/contexts/oci/repositories/ocireg.pushData({0x3963ee0, 0xc000141bc0}, {0x39492e0, 0xc0004fab40}, {{0x37ba313, 0x46}, {0xc00093ceb0, 0x47}, 0xc00, {0x0, ...}, ...}, ...)
	/Users/I500806/arbeit/ocm/pkg/contexts/oci/repositories/ocireg/utils.go:108 +0x2df
github.com/open-component-model/ocm/pkg/contexts/oci/repositories/ocireg.push({0x3963ee0, 0xc000141bc0}, {0x39492e0, 0xc0004fab40}, {0x3969fd8, 0xc00040dd60})
	/Users/I500806/arbeit/ocm/pkg/contexts/oci/repositories/ocireg/utils.go:98 +0x145
github.com/open-component-model/ocm/pkg/contexts/oci/repositories/ocireg.(*blobContainer).AddBlob(0xc000401740, {0x3969fd8, 0xc00040dd60})
	/Users/I500806/arbeit/ocm/pkg/contexts/oci/repositories/ocireg/blobs.go:111 +0x92
github.com/open-component-model/ocm/pkg/contexts/oci/repositories/ocireg.(*NamespaceContainer).AddBlob(0xc000063b60, {0x3969fd8, 0xc00040dd60})
	/Users/I500806/arbeit/ocm/pkg/contexts/oci/repositories/ocireg/namespace.go:139 +0xa3
github.com/open-component-model/ocm/pkg/contexts/oci/cpi.(*artefactBase).addBlob(0xc00040d400, {0x3969fd8, 0xc00040dd60})
	/Users/I500806/arbeit/ocm/pkg/contexts/oci/cpi/base.go:58 +0x4a
github.com/open-component-model/ocm/pkg/contexts/oci/cpi.(*ManifestImpl).AddBlob(0xc00040d400, {0x3969fd8, 0xc00040dd60})
	/Users/I500806/arbeit/ocm/pkg/contexts/oci/cpi/flavor_manifest.go:89 +0x45
github.com/open-component-model/ocm/pkg/contexts/ocm/repositories/genericocireg.(*StateAccess).Put(0xc0001fe040, {0xc0009d4800, 0x5a0, 0x800})
	/Users/I500806/arbeit/ocm/pkg/contexts/ocm/repositories/genericocireg/state.go:166 +0x31d
github.com/open-component-model/ocm/pkg/common/accessobj.(*state).Update(0xc00073c720)
	/Users/I500806/arbeit/ocm/pkg/common/accessobj/accessstate.go:295 +0x28b
github.com/open-component-model/ocm/pkg/contexts/ocm/repositories/genericocireg.(*ComponentVersionContainer).Update(0xc0000b6d20)
	/Users/I500806/arbeit/ocm/pkg/contexts/ocm/repositories/genericocireg/componentversion.go:193 +0x2d8
github.com/open-component-model/ocm/pkg/contexts/ocm/cpi/support.(*componentVersionAccessImpl).Update(0xc00080b0e0, 0x1)
	/Users/I500806/arbeit/ocm/pkg/contexts/ocm/cpi/support/compversaccess.go:391 +0x91
github.com/open-component-model/ocm/pkg/contexts/ocm/cpi/support.(*ComponentVersionAccess).Close(0xc000834630)
	/Users/I500806/arbeit/ocm/pkg/contexts/ocm/cpi/support/compversaccess.go:37 +0x54
github.com/open-component-model/ocm/pkg/common/accessio.(*once).Close(0xc00080b110)
	/Users/I500806/arbeit/ocm/pkg/common/accessio/utils.go:131 +0xaf
github.com/open-component-model/ocm/pkg/contexts/datacontext.(*sessionBase).Close(0xc0001b6190)
	/Users/I500806/arbeit/ocm/pkg/contexts/datacontext/session.go:141 +0xee
github.com/open-component-model/ocm/pkg/contexts/datacontext.(*session).Close(0xc0001b6190)
	/Users/I500806/arbeit/ocm/pkg/contexts/datacontext/session.go:92 +0xba
github.com/open-component-model/ocm/pkg/contexts/ocm.(*session).Close(0xc000499000)
	/Users/I500806/arbeit/ocm/pkg/contexts/ocm/session.go:81 +0x35
github.com/open-component-model/ocm/cmds/ocm/commands/ocmcmds/common/cmds/signing.(*SignatureCommand).Run(0xc00040d450)
	/Users/I500806/arbeit/ocm/cmds/ocm/commands/ocmcmds/common/cmds/signing/cmd.go:101 +0x63f
github.com/open-component-model/ocm/cmds/ocm/pkg/utils.SetupCommand.func1(0xc0008faf00, {0xc000390460, 0x1, 0x7})
	/Users/I500806/arbeit/ocm/cmds/ocm/pkg/utils/command.go:104 +0x1cb
github.com/spf13/cobra.(*Command).execute(0xc0008faf00, {0xc000198030, 0x7, 0x7})
	/Users/I500806/go/pkg/mod/github.com/mandelsoft/cobra@v1.4.1-0.20220816123314-fd3b703d78bf/command.go:875 +0xb27
github.com/spf13/cobra.(*Command).ExecuteC(0xc00026e280)
	/Users/I500806/go/pkg/mod/github.com/mandelsoft/cobra@v1.4.1-0.20220816123314-fd3b703d78bf/command.go:993 +0x8cb
github.com/spf13/cobra.(*Command).Execute(0xc00026e280)
	/Users/I500806/go/pkg/mod/github.com/mandelsoft/cobra@v1.4.1-0.20220816123314-fd3b703d78bf/command.go:921 +0x2f
main.main()
	/Users/I500806/arbeit/ocm/cmds/ocm/main.go:27 +0x3e
Process 6267 has exited with status 2
Detaching
dlv dap (6244) exited with code: 0


```
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know**:

**Environment**:

--- a/pkg/docker/pusher.go
+++ b/pkg/docker/pusher.go
@@ -124,7 +124,7 @@ func (p dockerPusher) push(ctx context.Context, desc ocispec.Descriptor, src res
 			err := remoteserrors.NewUnexpectedStatusErr(headResp)
 
 			var statusError remoteserrors.ErrUnexpectedStatus
-			if errors.As(err, statusError) {
+			if errors.As(err, &statusError) {
 				log.G(ctx).
 					WithField("resp", headResp).
 					WithField("body", string(statusError.Body)).
@@ -191,7 +191,7 @@ func (p dockerPusher) push(ctx context.Context, desc ocispec.Descriptor, src res
 			err := remoteserrors.NewUnexpectedStatusErr(resp)
 
 			var statusError remoteserrors.ErrUnexpectedStatus
-			if errors.As(err, statusError) {
+			if errors.As(err, &statusError) {
 				log.G(ctx).
 					WithField("resp", resp).
 					WithField("body", string(statusError.Body)).
@@ -277,7 +277,7 @@ func (p dockerPusher) push(ctx context.Context, desc ocispec.Descriptor, src res
 			err := remoteserrors.NewUnexpectedStatusErr(resp)
 
 			var statusError remoteserrors.ErrUnexpectedStatus
-			if errors.As(err, statusError) {
+			if errors.As(err, &statusError) {
 				log.G(ctx).
 					WithField("resp", resp).
 					WithField("body", string(statusError.Body)).

